activestorage/app/models/active_storage/blob.rb

```diff
@@ -230,6 +234,31 @@ def purge_later
     ActiveStorage::PurgeJob.perform_later(self)
   end
 
+  # Returns an instance of service, which can be configured globally or per attachment
+  def service
+    custom_service =
+      if service_name
+        configs = Rails.configuration.active_storage.service_configurations
+
+        begin
+          ActiveStorage::Service.configure service_name, configs
```

Comment:
This would build a new service instance for each blob–we need some kind of memoization here (`ServiceRegistry.fetch`?). That could also allow us to move exception handling logic there and make `#service` method much more readable:

```ruby
def service
  ActiveStorage::ServiceRegistry.fetch(service_name, self.class.service)
end
```

Reply:
I've added memoization, but it looks hard to move exception handling to the same place cause error messages are different enough